inc/commondropdown.class.php

```diff
@@ -272,271 +274,36 @@ private function addFilesFromRichText(): void {
-                  echo Html::getCheckbox([
-                     'title' => __('Clear'),
-                     'name'  => "_blank_".$field['name']
-                  ]);
-                  echo "&nbsp;".__('Clear');
-
-               } else {
-                  echo Html::file([
-                     'name'       => $field['name'],
-                     'onlyimages' => true,
-                  ]);
-               }
-               break;
-
-            case 'password':
-               echo "<input type='password' name='password' value='' size='20' autocomplete='new-password'>";
-               break;
-
-            case 'tinymce':
-               Html::textarea([
-                  'name'            => $field['name'],
-                  'value'           => $this->fields[$field['name']],
-                  'enable_richtext' => true,
-                  'enable_images'   => !($field['disable_images'] ?? false),
-               ]);
-               break;
-
-            case 'duration' :
-               $toadd = [];
-               for ($i=9; $i<=100; $i++) {
-                  $toadd[] = $i*HOUR_TIMESTAMP;
-               }
-               Dropdown::showTimeStamp($field['name'], [
-                  'min'             => 0,
-                  'max'             => 8*HOUR_TIMESTAMP,
-                  'value'           => $this->fields[$field['name']],
-                  'addfirstminutes' => true,
-                  'inhours'         => true,
-                  'toadd'           => $toadd
-               ]);
-               break;
-
-            case 'itemtypename':
-               $options = [
-                  'value'  => $this->fields[$field['name']]
-               ];
-
-               if (isset($field['itemtype_list'])) {
-                  $options['types'] = $CFG_GLPI[$field['itemtype_list']];
-               }
-
-               if (isset($options['types'])) {
-                  Dropdown::showItemTypes(
-                     $field['name'],
-                     $options['types'],
-                     $options
-                  );
-               }
-               return false;
-
-            default:
-               $this->displaySpecificTypeField($ID, $field);
-               break;
-         }
-         if (isset($field['unit'])) {
-            echo "&nbsp;".$field['unit'];
-         }
-
-         echo "</td></tr>\n";
-      }
 
-      if (isset($this->fields['is_protected']) && $this->fields['is_protected']) {
-         $options['candel'] = false;
-      }
+      echo TemplateRenderer::getInstance()->render('dropdown_form.html.twig', [
+         'item'   => $this,
+         'params' => $options,
+         'additional_fields' => $fields
+      ]);
+   }
 
-      if (isset($_REQUEST['_in_modal'])) {
-         echo "<input type='hidden' name='_in_modal' value='1'>";
-      }
-      $this->showFormButtons($options);
 
-      return true;
+   function getSpecificTypeField(int $ID, array $field): string {
+      return '';
```

Comment:
`displaySpecificTypeField` is used in some plugins. I guess it could be possible to handle backward compatibility with following code:
```php
function getSpecificTypeField(int $ID, array $field): string {
   if (method_exists($this, 'displaySpecificTypeField')) {
      Toolbox::deprecated();
      ob_start();
      $this->displaySpecificTypeField();
      return ob_get_clean();
   }
}

public function __call($name, $arguments) {
   if ($name === 'displaySpecificTypeField') {
      Toolbox::deprecated();
   }
}
```

Magic method `__call` will trigger a deprecation code for any plugin that would call this method directly. It is mandatory to remove definition of `displaySpecificTypeField` in current class to prevent `if (method_exists($this, 'displaySpecificTypeField'))` to be always evaluated to true.